src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocationsTests.cs

```diff
@@ -357,6 +357,59 @@ public static class Extensions
             return CS8VerifyAnalyzerAsync(originalCode);
         }
 
+        [Fact]
+        public Task CS_NoDiagnostic_ParametersDifferMoreThanOne()
+        {
+            string originalCode = @"
+using System;
+using System.Threading;
+class C
+{
+    void MyMethod(int i){}
+    void MyMethod(int i, bool b){}
+    void MyMethod(int i, bool b, CancellationToken c){}
+
+    public void M(CancellationToken ct)
+    {
+        MyMethod(1);
+    }
+}
+            ";
+            return CS8VerifyAnalyzerAsync(originalCode);
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_LambdaAndExtensionMethod_NoTokenInLambda()
+        {
+            // Only for local methods will we look for the ct in the top-most ancestor
+            // For anonymous methods we will only look in the immediate ancestor
+            string originalCode = @"
+using System;
+using System.Threading;
+public static class Extensions
+{
+    public static void Extension(this bool b, Action<int> action)
+    {
+    }
+    public static void MyMethod(this int i, CancellationToken c = default)
+    {
+    }
+}
+class C
+{
+    public void M(CancellationToken ct)
+    {
+        bool b = false;
+        b.Extension((j) =>
```

Comment:
Can you add an anonymous delegate version of this test? lambdas and delegates should almost always be treated equivalently, and I think this case is one of those.

Reply:
Done. For both C# and VB.